Digital Credentials: implement userAgentAllowsProtocol()

marcoscaceres opened the pull request:
#### 13d1709c2ffe886e469ecad874d12c06b91b2428
<pre>
Digital Credentials: implement userAgentAllowsProtocol()
<a href="https://rdar.apple.com/153776127">rdar://153776127</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=294352">https://bugs.webkit.org/show_bug.cgi?id=294352</a>

Reviewed by Anne van Kesteren.

This implements the userAgentAllowsProtocol() method, which allows
user agents to indicate whether they support a given protocol for
digital credentials. This is useful for sites to determine if they can
request a digital credential with a specific protocol.

Spec:
<a href="https://w3c-fedid.github.io/digital-credentials/#dom-digitalcredential-useragentallowsprotocol">https://w3c-fedid.github.io/digital-credentials/#dom-digitalcredential-useragentallowsprotocol</a>

WebKit only supports the &quot;org-iso-mdoc&quot; protocol, so this method
returns true only for that protocol.

Imports relevent Web Platform Tests. Upstream commit:

<a href="https://github.com/web-platform-tests/wpt/commit/c7fd2ce174cbc79b1e8d9b86c448d709a0e3f427">https://github.com/web-platform-tests/wpt/commit/c7fd2ce174cbc79b1e8d9b86c448d709a0e3f427</a>
Canonical link: <a href="https://commits.webkit.org/296786@main">https://commits.webkit.org/296786@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ed3f83f434129fdad3e22575f31aa8b20ae25e79

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/109390 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29048 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/19477 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/114595 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/59629 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/111353 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/29727 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/37636 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83139 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/112338 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23671 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/98527 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/63596 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23055 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/16670 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/59217 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/93039 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16712 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/117707 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/36431 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26966 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92147 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36802 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94789 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91962 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23460 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36893 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14639 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/32241 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/36324 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41800 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35996 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/39334 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/37699 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/Modules/identity/DigitalCredential.idl:
```diff
@@ -31,4 +31,5 @@
 ] interface DigitalCredential : BasicCredential {
     [SameObject] readonly attribute object data;
     readonly attribute IdentityCredentialProtocol protocol;
+    static boolean userAgentAllowsProtocol(DOMString protocol);
```
I usually put static members first.

marcoscaceres committed:
Digital Credentials: implement userAgentAllowsProtocol()
rdar://153776127
https://bugs.webkit.org/show_bug.cgi?id=294352

Reviewed by Anne van Kesteren.

This implements the userAgentAllowsProtocol() method, which allows
user agents to indicate whether they support a given protocol for
digital credentials. This is useful for sites to determine if they can
request a digital credential with a specific protocol.

Spec:
https://w3c-fedid.github.io/digital-credentials/#dom-digitalcredential-useragentallowsprotocol

WebKit only supports the "org-iso-mdoc" protocol, so this method
returns true only for that protocol.

Imports relevent Web Platform Tests. Upstream commit:

https://github.com/web-platform-tests/wpt/commit/c7fd2ce174cbc79b1e8d9b86c448d709a0e3f427
Canonical link: https://commits.webkit.org/296786@main
